pkg/apis/v2v/v1alpha1/virtualmachineimport_types.go

```diff
@@ -83,34 +80,110 @@ type VirtualMachineImportStatus struct {
+	MappingRulesChecking VirtualMachineImportConditionType = "MappingRulesChecking"
+
+	// Processing represents the status of the VM import process while in progress
+	Processing VirtualMachineImportConditionType = "Processing"
 )
 
-// VirtualMachineImportConditionType defines the condition of VM import
+// SucceededConditionReason defines the reasons for the Succeeded condition of VM import
 // +k8s:openapi-gen=true
-type VirtualMachineImportConditionType string
+type SucceededConditionReason string
 
-// These are valid conditions of of VM import.
+// These are valid reasons for the Succeeded conditions of VM import.
 const (
-	// Ready represents status of the VM import process being completed.
-	Ready VirtualMachineImportConditionType = "Ready"
+	// ValidationFailed represents a failure to validate the eligibility of the VM for import
+	ValidationFailed SucceededConditionReason = "ValidationFailed"
 
-	// Processing represents status of the VM import process while in progress
-	Processing VirtualMachineImportConditionType = "Processing"
+	// UpdatingSourceVMFailed represents a failure to stop VM or rename it on the source provider
+	UpdatingSourceVMFailed SucceededConditionReason = "UpdatingSourceVMFailed"
+
+	// VMCreationFailed represents a failure to create the VM entity
+	VMCreationFailed SucceededConditionReason = "VMCreationFailed"
+
+	// DataVolumeCreationFailed represents a failure to create data volumes based on source VM disks
+	DataVolumeCreationFailed SucceededConditionReason = "DataVolumeCreationFailed"
+
+	// VirtualMachineReady represents the completion of the vm import
+	VirtualMachineReady SucceededConditionReason = "VirtualMachineReady"
+)
+
+// ValidatingConditionReason defines the reasons for the Validating condition of VM import
+// +k8s:openapi-gen=true
+type ValidatingConditionReason string
+
+// These are valid reasons for the Validating conditions of VM import.
+const (
+	// ValidationCompleted represents the completion of the vm import resource validating
+	ValidationCompleted ValidatingConditionReason = "ValidationCompleted"
+
+	// SecretNotFound represents the nonexistence of the provider's secret
+	SecretNotFound ValidatingConditionReason = "SecretNotFound"
+
+	// MappingResourceNotFound represents the nonexistence of the mapping resource
+	MappingResourceNotFound ValidatingConditionReason = "MappingResourceNotFound"
+
+	// UnreachableProvider represents a failure to connect to the provider
+	UnreachableProvider ValidatingConditionReason = "UnreachableProvider"
+
+	// SourceVmNotFound represents the nonexistence of the source VM
+	SourceVMNotFound ValidatingConditionReason = "SourceVMNotFound"
+
+	// IncompleteMappingRules represents the inability to prepare the mapping rules
+	IncompleteMappingRules ValidatingConditionReason = "IncompleteMappingRules"
+
+	// MappingRulesViolated represents the violation of mapping rules
+	MappingRulesViolated ValidatingConditionReason = "MappingRulesViolated"
```

Comment:
What is the difference between `Validating/MappingRulesViolated` and `MappingRulesChecking.MappingRulesCheckingFailed`?

Reply:
`Validating/MappingRulesViolated` - the reason for failing the validation of import vm process is a blocking violation of mapping rules. That will cause the entire VMImport process to fail. the message in this context is concise.
In this case, we'll have also the condition `MappingRulesChecking` set to `False` with reason `MappingRulesCheckingFailed` and the message should list all of the violations.

If mapping rules check produces only warnings, `MappingRulesChecking` is set to 'True`, the reason will be `MappingRulesCheckingReportedWarnings` and the message of that condition should list all of the warnings.

That will guarantee that  even if `Validating` is true, the warnings remain part of the status, kept on `MappingRulesChecking` condition.

Reply:
I understand, thanks.
In that case it seems to me that `Validating/MappingRulesViolated` is superfluous if the failure listing will be in the `MappingRulesChecking` and its status will also be `False`.
What do you think about removing that reason from `Validating` and then  `Validating` will represents "technical issues or user input checks" and `MappingRulesChecking` will focus on the rules.

Reply:
Sure, we can do that the same way as we excluded the 'validating' from the 'processing' condition.